bfvmm/src/vmcs/test/test_vmcs_intel_x64.cpp

```diff
@@ -20,3 +21,220 @@
+    mocks.OnCall(state_in, vmcs_intel_x64_state::tr_limit).Do(tr_limit);
+
+    mocks.OnCall(state_in, vmcs_intel_x64_state::es_access_rights).Do(es_access_rights);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::cs_access_rights).Do(cs_access_rights);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ss_access_rights).Do(ss_access_rights);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ds_access_rights).Do(ds_access_rights);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ds_access_rights).Do(ds_access_rights);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::fs_access_rights).Do(fs_access_rights);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::gs_access_rights).Do(gs_access_rights);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ldtr_access_rights).Do(ldtr_access_rights);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::tr_access_rights).Do(tr_access_rights);
+
+    // Setup 64 bit state functions
+    mocks.OnCall(state_in, vmcs_intel_x64_state::cr0).Do(cr0);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::cr3).Do(cr3);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::cr4).Do(cr4);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::dr7).Do(dr7);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::rflags).Do(rflags);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::gdt_base).Do(gdt_base);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::idt_base).Do(idt_base);
+
+    mocks.OnCall(state_in, vmcs_intel_x64_state::es_base).Do(es_base);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::cs_base).Do(cs_base);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ss_base).Do(ss_base);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ds_base).Do(ds_base);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ds_base).Do(ds_base);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::fs_base).Do(fs_base);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::gs_base).Do(gs_base);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ldtr_base).Do(ldtr_base);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::tr_base).Do(tr_base);
+
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ia32_debugctl_msr).Do(ia32_debugctl_msr);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ia32_pat_msr).Do(ia32_pat_msr);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ia32_efer_msr).Do(ia32_efer_msr);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ia32_perf_global_ctrl_msr).Do(ia32_perf_global_ctrl_msr);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ia32_sysenter_cs_msr).Do(ia32_sysenter_cs_msr);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ia32_sysenter_esp_msr).Do(ia32_sysenter_esp_msr);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ia32_sysenter_eip_msr).Do(ia32_sysenter_eip_msr);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ia32_fs_base_msr).Do(ia32_fs_base_msr);
+    mocks.OnCall(state_in, vmcs_intel_x64_state::ia32_gs_base_msr).Do(ia32_gs_base_msr);
+
+    mocks.OnCall(state_in, vmcs_intel_x64_state::dump).Do(dump);
```

Comment:
In a later update (not this PR), I would write this as:

```
mocks.OnCall(state_in, vmcs_intel_x64_state::dump);
```

There really is no needed to emulate this function (it's for debugging). Instead, if by registering OnCall with no `.XXX()`, you are telling Hippomocks to do nothing.